packages/headless/src/features/folding/folding-slice.ts

```diff
@@ -2,8 +2,10 @@ import {createReducer} from '@reduxjs/toolkit';
 import {Result} from '../../api/search/search/result';
```

Comment:
This is probably the most important and difficult file to review, although it didn't change much since the last (closed) PR.

While reviewing it, having [this diagram](https://drive.google.com/drive/folders/14rJZ-FOXpjp14p_wiTblMro6woVWuUlu?usp=sharing) on the side might help.